pilot/pkg/model/gateway.go

```diff
@@ -461,6 +479,27 @@ func canMergeProtocols(current protocol.Instance, p protocol.Instance) bool {
 	return (current.IsHTTP() || current == p) && p.IsHTTP()
 }
 
+func canMergeServers(current []*networking.Server, gatewayNameByServer map[*networking.Server]string, server *networking.Server, gateway config.Config) bool {
+	// If strict namespace checking is disabled, allow merging servers from different namespaces
+	if !features.EnableStrictGatewayNamespaceChecking {
+		return true
+	}
+	if kube.IsInternalGatewayReference(gateway.Name) {
```

Comment:
I am not sure this function is safe. what if i name my gateway `foo-istio-autogenerated-k8s-gateway-bar` for example?

Note the only current usage of IsInternalGatewayReference is for `istioctl analyze`, which is not sensitive to bugs; security checks are very sensitive to these attacks